typing/env.ml

```diff
@@ -1993,7 +2048,11 @@ let add_components slot root env0 comps =
     cltypes;
     components;
     modules;
-  }
+  } in
+
+  let extension_constructors =
+    renormalize_ext env1 comps.comp_ext_constrs :: env0.extension_constructors in
+  { env1 with extension_constructors }
```

Comment:
I don't like the fact that this field is handled in a different place from all others here, this lacks regularity. I recommend inlining `renormalize_ext` along with the other definitions `let values ... let modules ...`.

Note: this function is not doing anything with `local_constraints` (the other Pathmap from the environment) and it seems fairly odd to me. Is it a bug?